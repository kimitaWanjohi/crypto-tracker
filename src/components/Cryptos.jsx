import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { fetchCryptos } from "../redux/cryptos/slice";

export default function Cryptos() {
  const dispatch = useDispatch();
  const cryptos = useSelector((state) => state.cryptos.cryptos);
  const loading = useSelector((state) => state.cryptos.loading);
  const error = useSelector((state) => state.cryptos.error);
  useEffect(() => {
    dispatch(fetchCryptos());
  }, [dispatch]);
  if (loading) return <h1 className="text-4xl">Loading...</h1>;
  if (error) return <h1 className="text-4xl container">{error}</h1>;
  return <h1 className="text-xl truncate">{JSON.stringify(cryptos)}</h1>;
}
