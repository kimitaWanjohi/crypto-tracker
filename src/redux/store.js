import { configureStore } from "@reduxjs/toolkit";
import cryptosReducer from "./cryptos/slice";

export default configureStore({
  reducer: {
    cryptos: cryptosReducer,
  },
});
