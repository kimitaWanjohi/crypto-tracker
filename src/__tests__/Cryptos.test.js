import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from '../redux/store';
import Cryptos from "../components/Cryptos";

describe("Cryptos", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Cryptos />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders a list of missions", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Cryptos />
        </Provider>
      )
      .toJSON();
    expect(tree.children.length).toBe(2);
  });
});
