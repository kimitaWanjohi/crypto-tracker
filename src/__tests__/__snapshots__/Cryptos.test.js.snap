// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cryptos renders correctly 1`] = `
<div
  className="w-full mx-auto min-h-screen dark:bg-gray-800 bg-white"
>
  <h1
    className="text-4xl dark:text-gray-900  my-2 font-bold text-center"
  >
    Cryptos
  </h1>
  <div
    className="container mx-auto px-4 mb-4"
  >
    <div
      className="w-full my-2 py-2"
    >
      <div
        className="w-full py-4 bg-gray-200 dark:bg-gray-700 rounded-lg shadow-md p-4 sticky top-40 z-5"
      >
        <input
          className="w-full bg-gray-200 dark:bg-gray-800  text-gray-500 rounded-lg shadow-md p-4"
          onChange={[Function]}
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <div
      className="grid grid-cols-2 md:grid-cols-2 lg:grid-cols-3 gap-4"
    />
  </div>
</div>
`;
